src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultNotificationDTO.java

```diff
@@ -111,6 +113,27 @@ public List<BuildLogEntry> extractBuildLogs(ProgrammingLanguage programmingLangu
         return buildLogEntries;
     }
 
+    @Override
+    public List<BuildJobDTOInterface> getBuildJobs() {
+        return getBuild().jobs().stream().map(job -> (BuildJobDTOInterface) job).toList();
+    }
+
+    @JsonIgnore
+    @Override
+    public List<StaticCodeAnalysisReportDTO> getStaticCodeAnalysisReports() {
+        List<StaticCodeAnalysisReportDTO> staticCodeAnalysisReports = new ArrayList<>();
+        getBuild().jobs().forEach(job -> staticCodeAnalysisReports.addAll(job.staticCodeAnalysisReports()));
+        return staticCodeAnalysisReports;
```

Comment:
Suggestion: `flatMap` does exactly what you want to do here. Can be left as-is as well, if you prefer.
If changed, the same can be applied in `getTestwiseCoverageReports()` below.

mutable list required:
```suggestion
        return getBuild().jobs.stream().flatMap(job -> job.staticCodeAnalysisReports().stream()).collect(Collectors.toCollection(ArrayList::new));
```
immutable list okay:
```suggestion
        return getBuild().jobs.stream().flatMap(job -> job.staticCodeAnalysisReports().stream()).toList();
```